checker/src/org/checkerframework/checker/initialization/InitializationAnnotatedTypeFactory.java

```diff
@@ -633,6 +635,24 @@ private void computeFieldAccessType(AnnotatedTypeMirror type,
         }
     }
 
+    /**
+     * Returns the annotations explicitly written on a constructor result.
+     * Callers should check that {@code constructorDeclaration} is in fact a declaration
+     * of a constructor.
+     *
+     * @param constructorDeclaration Declaration tree of constructor
+     * @return set of annotations explicit on the resulting type of the constructor
+     */
+    public Set<AnnotationMirror> getExplicitAnnotationsOnConstructorResult(MethodTree constructorDeclaration) {
```

Comment:
Good point.  I moved the method to AnnotationUtils.  InternalUtils has these methods which are similar:
````
List<? extends AnnotationMirror> annotationsFromTree(AnnotatedTypeTree node)
List<? extends AnnotationMirror> annotationsFromTree(TypeParameterTree node)
List<? extends AnnotationMirror> annotationsFromArrayCreation(NewArrayTree node, int level)  
````